recipes/boost/all/conanfile.py

```diff
@@ -141,25 +208,63 @@ def configure(self):
                 if not self.options.get_safe('without_%s' % lib):
                     raise ConanInvalidConfiguration("Boost '%s' library requires multi threading" % lib)
 
-        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime and self.options.shared:
+        if self.settings.compiler == "Visual Studio" and "MT" in str(self.settings.compiler.runtime) and self.options.shared:
             raise ConanInvalidConfiguration("Boost can not be built as shared library with MT runtime.")
 
+        # Check, when a boost module is enabled, whether the boost modules it depends on are enabled as well.
+        for mod_name, mod_deps in self._dependencies["dependencies"].items():
+            if not self.options.get_safe("without_{}".format(mod_name), True):
+                for mod_dep in mod_deps:
+                    if self.options.get_safe("without_{}".format(mod_dep), False):
+                        raise ConanInvalidConfiguration("{} requires {}: {} is disabled".format(mod_name, mod_deps, mod_dep))
+
+        if not self.options.without_python:
+            if not self.options.python_version:
+                self.options.python_version = self._detect_python_version()
+
     def build_requirements(self):
         if not self.options.header_only:
             self.build_requires("b2/4.2.0")
 
+    def _with_xxx(self, xxx):
```

Comment:
may we have more descriptive name than `xxx`?

Reply:
Because you asked so nicely :heart_eyes: